tools/ARIAtools/computeMisclosure.py

```diff
@@ -556,15 +559,15 @@ def __autoReferencePoint__(self):
 
             # Break loop after 10000 iterations
             if n == 10000:
-                print('WARNING: No reference point with coherence >= 0.7 found')
-                exit()
+                msg = f'No reference point with coherence >= {coh_min} found'
+                log.error(msg)
+                raise Exception(msg)
```

Comment:
@Askaholic would you mind suggesting more pythonic way to do this?

Reply:
Hmm, I guess you could raise the exception here and then catch it higher up the call stack. Something like:

```python3
def the_function():
    if foo:
        raise Exception("Error message")

def main():
    try:
        the_function()
    except Exception:
        log.exception("An unexpected error occurred")
        # Note that log.exception also prints the exception trace including the exception message
```

Reply:
@Askaholic won't this script continue to run if there are addition steps in main?

Reply:
I would assume you basically just wrap the whole body of main in the try block. Actually what would be even better is if you wrapped the call to main in a try block. Then any python callers can catch the exceptions and handle them in their own way:

```python

If __name__ == '__main__':
    try:
        main()
    except Exception:
        log.exception('ditto')
```

Reply:
@Askaholic yep, agreed - from my reading this seems to be the best practice